### PR TITLE
Addon Storysource improvements

### PR DESCRIPTION
**Pull request by igor-dv**
## What I did
- Use acorn to calculate locations instead of "line-column" pkg.
- Remove ugly comments from the storysource (currently only eslint things)
- Make "prettier" and "uglyComments" configurable in loader

### CODE REVIEW

File: addons/storysource/src/loader/generate-helpers.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    parts.push(start, end);
+    lastIndex = comment.end;
+  });
+
+  return parts.join('');
+}
+
+function prettifyCode(source) {
+  return prettier.format(source, {
```
Comment (Hypnosphi) [inline addons/storysource/src/loader/generate-helpers.js]: Can we make it optional and configurable?

Comment (igor-dv) [reply addons/storysource/src/loader/generate-helpers.js]: Only the prettier or the whole comments removal? Also. why?

Comment (Hypnosphi) [reply addons/storysource/src/loader/generate-helpers.js]: The prettier. Users may have their own preferences about how their code should be formatted

File: addons/storysource/src/loader/generate-helpers.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    stage3: true,
+    es7: true,
+  },
+};
+
+function generateSourceWithoutUglyComments(source, comments) {
+  let lastIndex = 0;
+  const parts = [source];
+
+  const isUgly = comment => comment.value.trim().indexOf('eslint-') === 0;
```
Comment (Hypnosphi) [inline addons/storysource/src/loader/generate-helpers.js]: also "global", "flow", "$FlowFixMe", probably others. Let's make it an array of regexp patterns, probably also configurable from outside

File: addons/storysource/src/loader/generate-helpers.js (hunk 1 above)
Comment (igor-dv) [reply addons/storysource/src/loader/generate-helpers.js]: Actually, I thought we won't expose it to users 😂 But I can make it configurable.

Comment (Hypnosphi) [reply addons/storysource/src/loader/generate-helpers.js]: > I thought we won't expose it to users

Why? I would like to use it in my project (and we don't use semicolons =)

Comment (igor-dv) [reply addons/storysource/src/loader/generate-helpers.js]: 👌

File: addons/storysource/src/loader/generate-helpers.js (hunk 2 above)
Comment (igor-dv) [reply addons/storysource/src/loader/generate-helpers.js]: I've added "global" and "eslint" as defaults. Do you think we need to support flow as well? I'm not familiar with its comments syntax.

Comment (Hypnosphi) [reply addons/storysource/src/loader/generate-helpers.js]: No, it's good defaults

File: addons/storysource/src/loader/generate-helpers.js [hunk 3]
```diff
@@ -20,33 +20,31 @@ const acornConfig = {
   },
 };
 
-function generateSourceWithoutUglyComments(source, comments) {
+function isUglyComment(comment, uglyCommentsRegex) {
+  return uglyCommentsRegex.find(regex => regex.test(comment));
```
Comment (Hypnosphi) [inline addons/storysource/src/loader/generate-helpers.js]: `.some` not `.find`

File: addons/storysource/src/loader/generate-helpers.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  locations: true,
+  plugins: {
+    jsx: true,
+    stage3: true,
+    es7: true,
+  },
+};
+
+function isUglyComment(comment, uglyCommentsRegex) {
+  return uglyCommentsRegex.find(regex => regex.test(comment));
```
Comment (Hypnosphi) [inline addons/storysource/src/loader/generate-helpers.js]: `.some` not `.find`

**Review by Hypnosphi — CHANGES_REQUESTED** [review]
Let's fix array method and update README — and we're good to go

**Review by Hypnosphi — APPROVED** [review]

**Comment by Hypnosphi** [comment]
```
#!/bin/bash -eo pipefail
yarn lint
yarn run v1.3.2
$ yarn lint:js . && yarn lint:ts **/*.ts && yarn lint:md .
$ cross-env NODE_ENV=production eslint --cache --cache-location=.cache/eslint --ext .js,.jsx,.json --report-unused-disable-directives .
$ tslint -p . -c tslint.json -t stylish '**/*.ts'
$ remark -q .
addons/storysource/README.md
  1:1  warning  Missing newline character at end of file  final-newline  remark-lint

⚠ 1 warning
Done in 41.52s.
```

Please fix remark warning

**Pull request merged**